core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -472,11 +473,19 @@ public IndexSettings getIndexSettings() {
      * {@link IndexReader}-specific optimizations, such as rewriting containing range queries.
      */
     public QueryShardContext newQueryShardContext(int shardId, IndexReader indexReader, LongSupplier nowInMillis) {
```

Comment:
That method is still being used in other places. For example in `MetaDataCreateIndexService` and `MetaDataIndexAliasesService`. In these places we don't have access to a search context and we test there if alias filters parse correctly, which means we will not be supporting `percolate` queries there, which I think is ok.